lte/gateway/c/core/oai/tasks/amf/amf_app_state_manager.cpp

```diff
@@ -156,18 +168,145 @@ amf_app_desc_t* AmfNasStateManager::get_state(bool read_from_redis) {
   state_dirty = true;
   if (persist_state_enabled && read_from_redis) {
     read_state_from_db();
+    read_ue_state_from_db();
+    amf_sync_app_maps_from_db();
   }
   return state_cache_p;
 }
 
-map_uint64_ue_context_t AmfNasStateManager::get_ue_state_map() {
-  return state_ue_map;
+map_uint64_ue_context_t* AmfNasStateManager::get_ue_state_map() {
+  return &state_ue_map;
+}
+
+/* Get the ue id from IMSI */
+bool get_amf_ue_id_from_imsi(
+    amf_app_desc_t* amf_app_desc_p, imsi64_t imsi64, amf_ue_ngap_id_t* ue_id) {
+  amf_ue_context_t* amf_ue_context_p = &amf_app_desc_p->amf_ue_contexts;
+  magma::map_rc_t rc_map             = magma::MAP_OK;
+  rc_map = amf_ue_context_p->imsi_amf_ue_id_htbl.get(imsi64, ue_id);
+  if (rc_map != magma::MAP_OK) {
+    return (false);
+  }
+  return true;
+}
+
+// This is a helper function for debugging. If the state manager needs to clear
+// the state in the data store, it can call this function to delete the key.
+void AmfNasStateManager::clear_db_state() {
+  OAILOG_DEBUG(LOG_AMF_APP, "Clearing state in data store");
+  std::vector<std::string> keys_to_del;
+  keys_to_del.emplace_back(AMF_NAS_STATE_KEY);
+
+  if (redis_client->clear_keys(keys_to_del) != RETURNok) {
+    OAILOG_ERROR(LOG_AMF_APP, "Failed to clear the state in data store");
+    return;
+  }
+}
+
+void put_amf_ue_state(
+    amf_app_desc_t* amf_app_desc_p, imsi64_t imsi64, bool force_ue_write) {
+  if (AmfNasStateManager::getInstance().is_persist_state_enabled()) {
+    if (imsi64 != INVALID_IMSI64) {
```

Comment:
For ease of readability

```suggestion
 if ((!AmfNasStateManager::getInstance().is_persist_state_enabled()) || (imsi64 == INVALID_IMSI64)) {
    return
  }
```

Reply:
Comment Addressed